sdk/core/azure-core/test/libcurl-stress-test/Dockerfile

```diff
@@ -3,11 +3,21 @@
 
 FROM mcr.microsoft.com/mirror/docker/library/ubuntu:22.04
 
-# copy the tagrget binary
-COPY ./bin/libcurl-stress-test/azure-core-libcurl-stress-test ./azure-core-libcurl-stress-test
-RUN chmod +x ./azure-core-libcurl-stress-test
-
+RUN su
 # install the mem check tool 
 RUN apt-get update -y
-RUN apt-get install valgrind -y
+RUN apt-get install gcc cmake make g++ git zip unzip build-essential pkg-config wget curl valgrind -y
+RUN wget -O vcpkg.tar.gz https://github.com/microsoft/vcpkg/archive/master.tar.gz
+RUN mkdir /opt/vcpkg
+RUN tar xf vcpkg.tar.gz --strip-components=1 -C /opt/vcpkg
+RUN /opt/vcpkg/bootstrap-vcpkg.sh
+RUN ln -s /opt/vcpkg/vcpkg /usr/local/bin/vcpkg
```

Comment:
Forgive me, but this is somewht terrifying... We're downloading arbitrary code from the internet and running it without even checking hashes to verify that we're getting the binary we expected.

Then we download the azure SDK and rebuild the libcurl stress test, which we will run, again without verifying what we downloaded.

IMHO you should be pulling a binary from an official build (probably using universal artifacts) similar to how we get the APIView tool from our official builds.

Downloading unknown, arbitrary code from the internet and then compiling and running it is not a good idea.

Reply:
How is this random code? 
What part of github  microsoft or vcpkg seem random? 
i know my web cam / mic might not be the best but do i look like a script kiddie that has no idea what they're doing? nothing is random there, every step has a purpose, 
if you would take five minutes to read the commands, they would make perfect sense. 

Allow me to elaborate : 
L9 -> install missing packages required for vcpkg and subsequent operations 
L10 -> wget the latest release from GITHUB under MICROSOFT organization VCPKG project 
L11 -> create folder under /opt to host vcpkg 
L12 -> extract the downloaded tar to it's destination and remove the first part of the name structure ( strip component)
L13 -> run bootstrap-vcpkg ( see vcpkg instructions ) 
L14 -> create a symbolic link in  usr/local/bin to  vcpkg under opt

Explain to me what is random here, what did i not understand, what did i put there for no reason whatsoever. 

Also how is that diffrent from  https://github.com/microsoft/vcpkg#quick-start-unix  ?
It's the same thing, wget is the equivalent of git clone , my way i get to control where things go more cleanly than git clone. 

And what is wrong with building the code?
That's what stress tests in go do https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/messaging/azservicebus/internal/stress/Dockerfile . 
That's what i was requested to do, i did not do it because i had nothing better to do with my life. 

Since you bring up that security point please show me where in the CI or LIVE or any other CPP pipeline we verify the hash of ANYTHING. such as the code, vcpkg, cahed components ? Let me save you some time , WE DO NOT. 
We choose to trust that github, and vcpkg are valid sources, that other tools we use are not malicious. 
Is this good or bad ? That's another discussion that if you want you can bring up with EngSys.  

Another point to bring up is you assume stress runs against a specific build that i can extract the binary from. That assumption is incorrect, it runs on it's own schedule, it runs on a weekly schedule. Thus i need to grab and build the code as needed, not based on some glorious build.  

I understand you are busy and don't have time to look and google all the commands, that's fine, i don't expect everyone to know everything, but assuming that i'm an imbecile who has no idea what they're doing i do find insulting.